eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/BomGenerator.java

```diff
@@ -144,6 +160,69 @@ private void generate() {
         return inputDependencies;
     }
 
+    private List<BomDependency> scan() {
+       var versioningClientDependency = scanVersioningClientFileDependencies();
+
+       if (this.overriddenInputDependenciesFileName == null) {
+           return versioningClientDependency;
+       }
+
+       var overriddenInputDependencies = scanOverriddenDependencies();
+       var overriddenInputDependenciesNoVersion = overriddenInputDependencies.stream().map(Utils::toBomDependencyNoVersion).collect(Collectors.toUnmodifiableSet());
+
+       var filteredInputDependencies = versioningClientDependency.stream().filter(dependency -> !overriddenInputDependenciesNoVersion.contains(toBomDependencyNoVersion(dependency))).collect(Collectors.toList());
+       filteredInputDependencies.addAll(overriddenInputDependencies);
+
+       return filteredInputDependencies;
+    }
+
+    private Map<BomDependency, String> parseRawFile() {
+        Map<BomDependency, String> inputDependencies = new HashMap();
+
+        try {
+            for (String line : Files.readAllLines(Paths.get(overriddenInputDependenciesFileName))) {
+                var matcher = INPUT_DEPENDENCY_PATTERN.matcher(line);
+                if (!matcher.matches()) {
+                    continue;
+                }
+
+                var dependencyPattern = matcher.group(1);
+                var pomFilePath = matcher.groupCount() == 3 ? matcher.group(2) : null;
+
+                var dependency = STRING_SPLIT_BY_COLON.split(dependencyPattern); {
+                    if(dependency.length != 3) {
+                        continue;
+                    }
+                    validateNotNullOrEmpty(dependency, "inputDependency");
+                    inputDependencies.put(new BomDependency(dependency[0], dependency[1], dependency[2]), pomFilePath);
+                }
+            }
+        } catch (IOException exception) {
+            logger.error("Input file parsing failed. Exception{}", exception.toString());
+        }
+
+        return inputDependencies;
+    }
+
+    private List<BomDependency> scanOverriddenDependencies() {
+        List<BomDependency> allInputDependencies = new ArrayList<>();
+
+        var overriddenInputDependencies = parseRawFile();
+        // Some of these libraries may not have been published yet.
```

Comment:
What is the scenario where we use a dependency that is not published to maven?

Reply:
At this point none - but I can think of a scenario where someone wants to test a beta library to plan for work ahead of time. Since, we were adding this functionality I included both.